src/control/server/storage/bdev/backend_test.go

```diff
@@ -422,74 +421,113 @@ func TestBdev_Backend_Update(t *testing.T) {
 	}
 }
 
-func TestBdev_Backend_cleanHugePages(t *testing.T) {
-	curUsr, _ := user.Current()
+type mockFileInfo struct {
+	name    string
+	size    int64
+	mode    os.FileMode
+	modTime time.Time
+	isDir   bool
+	stat    *syscall.Stat_t
+}
+
+func (mfi *mockFileInfo) Name() string       { return mfi.name }
+func (mfi *mockFileInfo) Size() int64        { return mfi.size }
+func (mfi *mockFileInfo) Mode() os.FileMode  { return mfi.mode }
+func (mfi *mockFileInfo) ModTime() time.Time { return mfi.modTime }
+func (mfi *mockFileInfo) IsDir() bool        { return mfi.isDir }
+func (mfi *mockFileInfo) Sys() interface{}   { return mfi.stat }
+
+func testFileInfo(t *testing.T, name string, uid uint32) os.FileInfo {
+	t.Helper()
+
+	return &mockFileInfo{
+		name: name,
+		stat: &syscall.Stat_t{
+			Uid: uid,
+		},
+	}
+}
+
+type testWalkInput struct {
+	path   string
+	info   os.FileInfo
+	err    error
+	expErr error
+}
+
+func TestBdev_Backend_cleanHugePagesFn(t *testing.T) {
+	testDir := "/wherever"
 
 	for name, tc := range map[string]struct {
-		owner, tgtUsr, filePrefix, lookPrefix string
-		nrPages, expNrRemoved, expNrRemain    int
-		expErr                                error
+		prefix     string
+		tgtUid     string
+		testInputs []*testWalkInput
+		expRemoved []string
 	}{
-		"success": {
-			owner:        curUsr.Username,
-			tgtUsr:       curUsr.Username,
-			filePrefix:   "spdk_",
-			lookPrefix:   "spdk",
-			nrPages:      10,
-			expNrRemoved: 10,
-			expNrRemain:  0,
+		"ignore subdirectory": {
```

Comment:
How about testcases for the following:
  * err received from `Walk()`
  * nil value returned by `FileInfo.Sys()`
  * err received by `remove()`

I think that should give you 100% coverage for the closure and will ensure that future changes don't regress error handling for those scenarios.

Reply:
Agreed